src/app/modules/item/services/item-task.service.ts

```diff
@@ -0,0 +1,120 @@
+  private error$ = merge(
+    this.answerService.error$,
+    this.viewsService.error$,
+  ).pipe(switchMap(error => throwError(() => error)));
+
+  readonly task$ = merge(this.initService.task$, this.error$);
+  readonly iframeSrc$ = this.initService.iframeSrc$;
+  get initialized(): boolean {
+    return this.initService.initialized;
+  }
+
+  readonly views$ = this.viewsService.views$;
+  readonly display$ = this.viewsService.display$;
+  readonly activeView$ = this.viewsService.activeView$;
+
+  private config$ = this.initService.config$;
+
+  constructor(
+    private initService: ItemTaskInitService,
+    private answerService: ItemTaskAnswerService,
+    private viewsService: ItemTaskViewsService,
+    private itemRouter: ItemRouter,
+    private itemNavigationService: ItemNavigationService,
+    private router: Router,
+    private localeService: LocaleService,
+  ) {}
+
+  configure(route: FullItemRoute, url: string, attemptId: string): void {
+    this.initService.configure(route, url, attemptId);
+  }
+
+  initTask(iframe: HTMLIFrameElement): void {
+    this.initService.initTask(iframe, task => this.bindPlatform(task));
+  }
+
+  showView(view: string): void {
+    this.viewsService.showView(view);
+  }
+
+  private bindPlatform(task: Task): void {
+    const platform = new TaskPlatform({
+      validate: (mode): Observable<void> => this.validate(mode).pipe(mapTo(undefined)),
+      getTaskParams: (): Observable<TaskParamsValue> =>
+        of({ minScore: -3, maxScore: 10, randomSeed: 0, noScore: 0, readOnly: false, options: {} }),
+      updateDisplay: (display): Observable<void> => {
+        this.viewsService.updateDisplay(display);
+        return EMPTY;
+      },
+      showView: (view): Observable<void> => {
+        this.viewsService.showView(view);
+        return EMPTY;
+      },
+      openUrl: (params): Observable<void> => {
+        if (typeof params === 'string') return this.navigateToItem(params);
+        if ('path' in params) return this.navigateToItem(params.path, params.newTab);
+        this.navigate(params.url, params.newTab);
+        return EMPTY;
+      }
+    });
+    task.bindPlatform(platform);
+  }
+
+  private validate(mode: string): Observable<unknown> {
+    switch (mode) {
+      case 'cancel': return this.answerService.reloadAnswer();
+      case 'nextImmediate': return this.navigateToNextItem();
+      case 'next': return this.answerService.submitAnswer().pipe(switchMap(() => this.navigateToNextItem()));
+      case 'top': return this.answerService.submitAnswer().pipe(switchMap(() => this.scrollTop()));
+      default: return this.answerService.submitAnswer();
+    }
+  }
+
+  private navigateToNextItem(): Observable<void> {
+    return this.config$.pipe(
+      take(1),
+      switchMap(({ route }) => this.itemNavigationService.getNavigationNeighbors(route)),
+      tap(data => {
+        if (data.right) this.itemRouter.navigateTo(data.right);
+      }),
+      mapTo(undefined),
+    );
+  }
+
```

Comment:
Working. But so low readibility to such a simple operation :-( . I have no suggestion of improvements for the moment.

Reply:
:shrug: